### PR TITLE
feat: ``KeyValues`` for entity input

### PR DESCRIPTION
**Pull request by Kxnrl**
port from modsharp.
cs2fixes need more tests.

### support input
- KeyValues

### support entity key values
- targetname
- origin
- angles
- max_health
- ~~solid~~ broken in cs2
- movetype
- EntityTemplate
- basevelocity
- target
- filtername
- force

### CODE REVIEW

File: src/customio.cpp [hunk 1]
```diff
@@ -0,0 +1,242 @@
+#include "customio.h"
```
Comment (xen-000) [inline src/customio.cpp:1]: Add the license notice here, you've already done it in the header too.

File: src/customio.cpp [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    AddOutputHandler_t m_Handler;
+};
+
+static void AddOutputCustom_Targetname(Z_CBaseEntity*                  pInstance,
+                                       CEntityInstance*                pActivator,
+                                       CEntityInstance*                pCaller,
+                                       const std::vector<std::string>& vecArgs)
+{
+    addresses::CEntityIdentity_SetEntityName(pInstance->m_pEntity, vecArgs[1].c_str());
+    Message("SetName %s to %d", vecArgs[1].c_str(), pInstance->GetHandle().GetEntryIndex());
```
Comment (xen-000) [inline src/customio.cpp:54]: Probably best to put these logs behind `#ifdef _DEBUG`, this applies to all the other handlers.

File: src/customio.cpp [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    const int m_iMaxHealth = pInstance->m_iMaxHealth;
+    Message("SetMaxHealth %d for %s", m_iMaxHealth, pInstance->GetName());
+}
+
+static void AddOutputCustom_Health(Z_CBaseEntity*                  pInstance,
+                                   CEntityInstance*                pActivator,
+                                   CEntityInstance*                pCaller,
+                                   const std::vector<std::string>& vecArgs)
+{
+    const auto value = clamp(Q_atoi(vecArgs[1].c_str()), 1, 99999999);
```
Comment (xen-000) [inline src/customio.cpp:96]: IIRC some maps out there can set health to 0 or -1, not to mention there's already the `SetHealth` input.

File: src/customio.cpp [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+}
+
+static void AddOutputCustom_BaseVelocity(Z_CBaseEntity*                  pInstance,
+                                         CEntityInstance*                pActivator,
+                                         CEntityInstance*                pCaller,
+                                         const std::vector<std::string>& vecArgs)
+{
+    const Vector velocity(clamp(Q_atof(vecArgs[1].c_str()), -3500.f, 3500.f),
+                          clamp(Q_atof(vecArgs[2].c_str()), -3500.f, 3500.f),
+                          clamp(Q_atof(vecArgs[3].c_str()), -3500.f, 3500.f));
```
Comment (xen-000) [inline src/customio.cpp:142]: This should be clamped to 4096 since that's the actual limit of velocity vectors in CS2.

**Review by xen-000 — CHANGES_REQUESTED** [review]

File: src/customio.cpp (hunk 3 above)
Comment (Kxnrl) [reply src/customio.cpp:96]: add health 0 to pawn will cause some bug.
I think we also can proxy to ``SetHealth``

File: src/customio.cpp (hunk 4 above)
Comment (Kxnrl) [reply src/customio.cpp:142]: that old values are came from cs:go.

**Comment by Kxnrl** [comment]
@xen-000 btw, the copytight is 2023. update all to 2024 after merge :bug:.

**Review by xen-000 — APPROVED** [review]

**Pull request merged**